rmgpy/cantherm/statmech.py

```diff
@@ -772,6 +792,41 @@ def applyEnergyCorrections(E0, modelChemistry, atoms, bonds,
 
     return E0
 
+class Log(object):
+    """
+    Represent a general log file.
+    The attribute `path` refers to the location on disk of the log file of interest.
+    A method is provided to determine whether it is a Gaussian, Molpro, or QChem type.
+    """
+
+    def __init__(self, path):
+        self.path = path
+
+    def determine_qm_software(self, fullpath):
+        """
+        Given a path to the log file of a QM software, determine whether it is Gaussian, Molpro, or QChem
+        """
+        f = open(fullpath, 'r')
+        line = f.readline()
+        software = None
+        while line != '':
+            if 'gaussian' in line.lower():
+                f.close()
+                software = GaussianLog(fullpath)
+                break
+            elif 'qchem' in line.lower():
+                f.close()
+                software = QchemLog(fullpath)
+                break
+            elif any(x in line for x in ['ATOMIC COORDINATES','PROGRAM']):
```

Comment:
At least for Molpro 2015, it prints `MOLPRO` towards the top of the file. Should we check for that instead?